tracer/src/Datadog.Trace/RemoteConfigurationManagement/Subscription.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="Subscription.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.RemoteConfigurationManagement
+{
+    internal class Subscription
+    {
+        private readonly Action _onChange;
+
+        public Subscription(Func<Dictionary<string, List<RemoteConfiguration>>, Dictionary<string, List<RemoteConfigurationPath>>?, List<ApplyDetails>> callback, Action onChange, params string[] productKeys)
+        {
+            _onChange = onChange;
+            ProductKeys = productKeys.ToList();
+            Callback = callback;
+        }
+
+        internal List<string> ProductKeys { get; set; }
+
+        internal Func<Dictionary<string, List<RemoteConfiguration>>, Dictionary<string, List<RemoteConfigurationPath>>?, List<ApplyDetails>> Callback { get; set; }
+
+        internal List<RemoteConfiguration>? NextConfigs { get; set; }
+
+        public void UnsubscribeProducts(params string[] names)
```

Comment:
I can't find any usage for these, but I would say that's a good thing. A subscription would be easier to understand if it's immutable. Could we remove the possibility to dynamically add or remove products?

If it's necessary to change the products that are subscribed to, we could always recreate the subscription.

Reply:
Actually, I'm wrong this is used. I think making `Subscription` *could* be a good idea, probably best to merge as is and experiment later.

Reply:
yes I could make it immutable, I was just wondering if it was potentially less performant, as it implies removing the old one from the list, taking into account the keys of the old one to concatenate to the new keys, and adding the new subscription in the list but maybe it's not that much more

Reply:
Made the Subscription immutable and removed all this. When we add or remove products, a new instance is provided